stock_orderpoint_route/models/stock_warehouse_orderpoint.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2019 Eficent Business and IT Consulting Services, S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class StockWarehouseOrderpoint(models.Model):
+    _inherit = 'stock.warehouse.orderpoint'
+
+    route_ids = fields.Many2many(
+        'stock.location.route', string='Allowed routes',
+        compute='_compute_route_ids',
+    )
+    route_id = fields.Many2one('stock.location.route', string='Route',
+                               domain="[('id', 'in', route_ids)]",
+                               ondelete='restrict')
+
+    @api.depends('product_id', 'warehouse_id', 'location_id')
+    def _compute_route_ids(self):
+        route_obj = self.env['stock.location.route']
+        for wh in self.mapped('warehouse_id'):
+            wh_routes = route_obj.search(
+                [('warehouse_ids', '=', wh.id)])
```

Comment:
Is there a reason to filter on the warehouse if anyway we filter on the location afterwards?

Reply:
Instead of filtering on "warehouse selectable" + sublocations, do you think we can filter on a new flag "orderpoint selectable" + sublocations?

The end result will be the same as anyway we filter on the location of the orderpoint, but it seems more understandable to me.

I'm moving forward on the migration to 12.0, so it's the best time for me to make a change, that I can backport if you agree.

cc @jgrandguillaume

Reply:
@jbeficent I must say I agree with @guewen here, do you?

Reply:
I overlooked the "look in the routes of the product, look in the routes of the product categories, look in the routes of the warehouse" part. Still, I would be interested to have your thoughts on this @jbeficent.